qa/workunits/rbd/rbd_mirror.sh

```diff
@@ -124,20 +150,29 @@ remove_image_retry ${CLUSTER2} ${POOL} ${image1}
 testlog "TEST: test image rename"
 new_name="${image}_RENAMED"
 rename_image ${CLUSTER2} ${POOL} ${image} ${new_name}
+if [ "${RBD_MIRROR_MODE}" = "snapshot" ]; then
+  mirror_image_snapshot ${CLUSTER2} ${POOL} ${new_name}
+fi
 wait_for_image_replay_started ${CLUSTER1} ${POOL} ${new_name}
 wait_for_status_in_pool_dir ${CLUSTER1} ${POOL} ${new_name} 'up+replaying'
 admin_daemons ${CLUSTER1} rbd mirror status ${POOL}/${new_name}
 admin_daemons ${CLUSTER1} rbd mirror restart ${POOL}/${new_name}
 wait_for_image_replay_started ${CLUSTER1} ${POOL} ${new_name}
 wait_for_status_in_pool_dir ${CLUSTER1} ${POOL} ${new_name} 'up+replaying'
 rename_image ${CLUSTER2} ${POOL} ${new_name} ${image}
+if [ "${RBD_MIRROR_MODE}" = "snapshot" ]; then
+  mirror_image_snapshot ${CLUSTER2} ${POOL} ${image}
+fi
 wait_for_image_replay_started ${CLUSTER1} ${POOL} ${image}
 
 testlog "TEST: test trash move restore"
 image_id=$(get_image_id ${CLUSTER2} ${POOL} ${image})
 trash_move ${CLUSTER2} ${POOL} ${image}
 wait_for_image_present ${CLUSTER1} ${POOL} ${image} 'deleted'
 trash_restore ${CLUSTER2} ${POOL} ${image_id}
+if [ "${RBD_MIRROR_MODE}" = "snapshot" ]; then
+  enable_mirror ${CLUSTER2} ${POOL} ${image} ${RBD_MIRROR_MODE}
```

Comment:
I think the condition like this would look confusing for an occasional reader. It would suggest that for the snapshot based mirroring trash restore does not restore mirroring while for the journal it does. While what actually happens here is that when testing journaling mode, the pool mirroring mode is "pool", while testing snapshot it is "image".

I think one approach would be to hide this condition in `trash_restore`, where you could use `MIRROR_POOL_MODE` in the condition and it would not be confusing.

Another, more radical approach would be to change the "journal" tests to set the pool ${POOL} mirroring mode to "image", similarly to "snapshot" tests. Then we could have less confusing conditions like this one. And for "journal" just add yet another mirror pool in pool mode and have tests that run for "journal" only, that just check that the mode works as expected. The "pool" mode is considered "deprecated" anyway. I would like to see what @idryomov  thinks about it.

Reply:
> I think one approach would be to hide this condition in trash_restore, where you could use MIRROR_POOL_MODE in the condition and it would not be confusing.

This looks optimistic to me. I will change it to the later one if Ilya insist.